packages/core/src/consumed-thing.ts

```diff
@@ -555,7 +557,32 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
         form = this.handleUriVariables(tp, form, options);
 
         const content = await client.readResource(form);
-        return new InteractionOutput(content, form, tp);
+        return this.handleInteractionOutput(content, form, tp);
+    }
+
+    private handleInteractionOutput(
+        content: Content,
+        form: TD.Form,
+        outputDataSchema: WoT.DataSchema | undefined
+    ): InteractionOutput {
+        // infer media type from form if not in response metadata
+        if (!content.type) content.type = form.contentType ?? "application/json";
```

Comment:
I think this *could* be shortened very slightly, like so:

```suggestion
        content.type ??= form.contentType ?? "application/json";
```

However, should the `type` within `content` be overwritten here? As the `content` is not returned again by this method, maybe we could consider something like this instead:

```suggestion
        const contentType = content.type ?? form.contentType ?? "application/json";
```